### PR TITLE
Launch macOS application with NSWorkspace and use OSLogStore for log retrieval

### PR DESCRIPTION
**Pull request by jwerle**
This was a wild one, but I tried to leave as many comments as possible!

### CODE REVIEW

**Comment by chicoxyzzy** [comment]
I see this in my terminal
```
# fs.watch
# fs.write
# fs.writeFile
2023-02-24 05:16:37.711 ssc[66872:8775122] error: OSLogStore: The log archive is corrupt or incomplete and cannot be read
```
and then I don't see any log output

**Comment by jwerle** [comment]
> I see this in my terminal
> 
> ```
> # fs.watch
> # fs.write
> # fs.writeFile
> 2023-02-24 05:16:37.711 ssc[66872:8775122] error: OSLogStore: The log archive is corrupt or incomplete and cannot be read
> ```
> 
> and then I don't see any log output

It should be fixed now!

**Comment by mribbons** [comment]
I'm just going to say: It's a lot of platform specific code 😄

**Review by chicoxyzzy — APPROVED** [review]
it works for me now

**Comment by chicoxyzzy** [comment]
BTW @jwerle maybe you can take a look at https://github.com/socketsupply/socket/pull/189 as well? I'm stuck there

File: test/src/process.js [hunk 1]
```diff
@@ -3,6 +3,9 @@ import process from 'socket:process'
 import { primordials } from 'socket:ipc'
 import path from 'path-browserify'
 
+// make `path-browserify` happy
+globalThis.process = process
```
Comment (chicoxyzzy) [inline test/src/process.js:7]: We can also delete it from `globalThis` at the and of the module

Comment (chicoxyzzy) [reply test/src/process.js:7]: actually, we need `path.resolve` in the `socket:path` :)

Comment (jwerle) [reply test/src/process.js]: haha yes TODO!

**Comment by jwerle** [comment]
> BTW @jwerle maybe you can take a look at #189 as well? I'm stuck there

we'll have to think about our approach here as using `OSLogStore` won't work

**Pull request merged**